### DISCUSSION TITLE
Statistics of the notebook

### DISCUSSION THREAD

**Discussion by Nick222** [zim-desktop-wiki/zim-desktop-wiki#1735] [Q&A]
Can the user see the statistics of the notebook as a whole - at least the number of notes in it?

**Comment by introt** [comment]
While Zim doesn't expose much through the GUI, given the notes are just text files, this can be accomplished by eg. standard UNIX tools; running `find -name '*.txt' | wc -l` inside your notebook directory gets you the number of `.txt` files, while something like
```sh
find -type f -name '*.txt' -exec grep -m 1 '^Content-Type: text/x-zim-wiki$' {} + | wc -l
```
should only count notes. Since we're using `find`, we can add arguments like `Journal` and `-mtime -30` to limit the search. Add `-l` to `grep` and remove the pipe to `wc`, and we get
```
find Journal -mtime -30 -type f -name '*.txt' -exec grep -l -m 1 '^Content-Type: text/x-zim-wiki$' {} +
```
which prints you a list of Journal pages modified in the last 30 days (See also Search > Recent Changes... in the GUI).

**Reply by introt** [reply]
Now that the Zeitgeist plugin has been fixed, it can also be used to provide a number of statistics. Your version control system might also hold some insights, if you're using one.

**Comment by Nick222** [comment]
It is important for me to find those text files that Zim himself does not include in his index.
I think this can only be done from within Zim.
